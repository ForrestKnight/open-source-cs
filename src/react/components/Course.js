/* Packages */
import React from 'react';


/* Component */
export class Course extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div className="Course">
                <a 
                    className="Course__name"
                    href={this.props.name.link}
                >
                    <h3>{this.props.name.text}</h3>
                </a>

                <p className="Course__school">
                    {this.props.school}
                </p>

                <p className="Course__duration">
                    {this.props.duration} weeks
                </p>

                <p className="Course__effort">
                    {this.props.effort} hours/week
                </p>

                <p className="Course__frequency">
                    {this.props.frequency}
                </p>

                {this.props.prerequisites &&
                    <div className="Course__prerequisites">
                        <p>{this.props.prerequisites}</p>
                    </div>
                }
            </div>
        );
    }
}


export default Course;