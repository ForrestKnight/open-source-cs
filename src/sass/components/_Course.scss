.Course {
    background: lighten($white, 5%);
    width: 300px;
    margin: 0 auto;
    text-align: center;
    display: block;
    padding: 8px;
    text-decoration: none;
    margin-bottom: 16px;

    @media #{$tablet_breakpoint} {
        width: 400px;
        padding: 16px;
    }

    
    @media #{$desktop_breakpoint} {
        display: inline-block;
        padding: 0 16px;
        padding-bottom: 16px; 
        margin: 0 20px;
        margin-bottom: 32px;
    }
}


.Course__school {
    @include courseDetail;
}


.Course__duration {
    @include courseDetail;
}


.Course__effort {
    @include courseDetail;
}


.Course__frequency {
    @include courseDetail;
}


.Course__prerequisites {
    @include courseDetail;
}