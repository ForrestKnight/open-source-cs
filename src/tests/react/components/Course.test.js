/* Packages */
import React from 'react';
import {shallow} from 'enzyme';


/* Component */
import {Course} from '../../../react/components/Course';


/* Test Config */
let component;
beforeEach(() => {
    component = shallow(<Course />);
});


/* Tests */
test('Render Course component', () => {
    expect(component).toMatchSnapshot();
});

