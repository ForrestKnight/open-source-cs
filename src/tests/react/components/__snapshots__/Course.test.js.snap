// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Course component with prerequisites 1`] = `
<div
  className="Course"
>
  <a
    className="Course__name"
    href="Link to course"
  >
    <h3>
      Name of course
    </h3>
  </a>
  <p
    className="Course__school"
  >
    School
  </p>
  <p
    className="Course__duration"
  >
    Duration
     weeks
  </p>
  <p
    className="Course__effort"
  >
    Effort
     hours/week
  </p>
  <p
    className="Course__frequency"
  >
    Frequency
  </p>
  <div
    className="Course__prerequisites"
  >
    <p>
      Prerequisite 1
      Prerequisite 2
    </p>
  </div>
</div>
`;

exports[`Render Course component without prerequisites 1`] = `
<div
  className="Course"
>
  <a
    className="Course__name"
    href="Link to course"
  >
    <h3>
      Name of course
    </h3>
  </a>
  <p
    className="Course__school"
  >
    School
  </p>
  <p
    className="Course__duration"
  >
    Duration
     weeks
  </p>
  <p
    className="Course__effort"
  >
    Effort
     hours/week
  </p>
  <p
    className="Course__frequency"
  >
    Frequency
  </p>
</div>
`;
